enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/readreplica/EnterpriseReadReplicaEditionModule.java

```diff
@@ -381,6 +382,8 @@ protected DuplexPipelineWrapperFactory pipelineWrapperFactory()
     public void registerEditionSpecificProcedures( Procedures procedures ) throws KernelException
     {
         procedures.registerProcedure( EnterpriseBuiltInDbmsProcedures.class, true );
+        // todo should readReplicas have access to those procedures?
+        procedures.registerProcedure( EnterpriseBuiltInProcedures.class, true );
```

Comment:
@martinfurmanski What do you think about this fellow? Should we register procedures that can write on read replicas? I assume there are other mechanism working to prevent RR from actually starting those kind of transactions.

Reply:
@burqen Probably better not to register it at all to avoid any issues. Check with the procedures- and security team otherwise, because the procedure execution and security model is not really an area I know well.

Reply:
I'll try calling a friend. Maybe @fickludd can give some pointers?

Reply:
I would have just gone with 50/50, but I'm a gambler.

Reply:
`org.neo4j.kernel.impl.api.KernelTransactionImplementation#schemaWrite` asserts that the instance has write capabilities, and will crash if it doesn't. Dunno if you think the error message is good enough or you want something more precise, but it shouldn't mess up Neo4j at least. 

I don't remember ever seeing us registering different procs depending on read/write capabilities before.

Reply:
So the behaviour should be pretty much the same as if you tried to issue a normal write query to the RR then. I think that is fine and it make sense to me to register this class of procedures to RR. When we later add procedures to this class that might not be write procedures, we don't want to spend the time tracking down why the procedures are not registered.
Thank you gents!